dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/HostConnectionPoolImpl.java

```diff
@@ -355,7 +351,19 @@ public boolean closeConnection(Connection<CL> connection) {
 				monitor.incConnectionClosed(host, connection.getLastException());
 			}
 		}
-		
+
+		@Override
+		public void recycleConnection(Connection<CL> connection) {
+			try {
+				this.closeConnection(connection);
+				// Create a new connection and add it to pool
+				createConnectionWithRetries();
```

Comment:
You need to check if this succeeded or not, and take action accordingly.

Reply:
Done